### ISSUE TITLE
[Spell][Talent][Priest] Abolish Disease behaviour with Body and Soul

### ISSUE THREAD

**Issue by sherndy** [dalaranwow/dalaran-wow#5642]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: When you cast Abolish Disease on self, powered up by Body and Soul talent (from holy tree) it only attempts to remove poison on the first tick

**Expected behaviour**: It should remove poison on every tick

**Steps to reproduce the problem**:

1. Cast it on yourself with the talent speced
2. ?????????????
3. profit

**Include proofs for this behaviour**

https://www.wowhead.com/spell=64129/body-and-soul#comments:id=959626:reply=445227
wowhead comment

http://www.arenajunkies.com/topic/46337-macro-abolish-n-cure-disease/
3rd post on the thread
"Yes but you're going to remove a lot more poison/disease if you leave abolish up and spam remove."
By context, it clearly implies not reseting the abolish tick timers increases poison removal, implying the ticks remove poison

--- From this point on there are no "hard" proofs ---

I promise I tried to find youtube videos of people casting it (since I have the foundest memories of playing holy/disc hybrid and kiting rogues cause of this spell - by nullifying crippling poison) on self on bgs and arenas, but you'd be surprised at how bad people are, most dont even have the spell on their bars, the ones that do, dont have it bound, and the rare cases that bind it didn't cast it on video.

About the interpretation of the tooltip that says: "When you cast Power Word: Shield or Leap of Faith, you increase the target's movement speed by 30/60% for 4 sec, and you have a 50/100% chance when you cast Abolish Disease on yourself to also cleanse 1 poison effect in addition to diseases."
Goes without saying that blizzard often poorly words it's descriptions (google hearthstone cards text vs behaviour) but I'd argue the part that says "when you cast Abolish Disease on yourself" is to be interpreted with a semantic focus on "on yourself" and not "you cast Abolish Disease", this whole section of the phrase serves the purpose of setting up who's the target when the talent takes effect. It is not however meant to point out WHEN the talent takes effect. Also in an analogy, just like Cure Disease gains the bonus added effect from druids' "Cure Poison" stands to reason Abolish Disease will gain the effect from druids' "Abolish Poison".

I'm aware memory plays tricks on your mind and all the psychology on people fervorly believe things that didn't happen at all. But I've played retail on late WotLK and this talent, and it's effect on both PW:S and Abolish, were what enabled me to kite rogues and dk's on a instant cast heavy holy-disc hybrid, that which enabled me to play holy on pvp at all, otherwise I'd spend most my life getting melted by ArP classes such as DK, Warrior, and rogues. Having a talent that single-handedly deals with two out of my 3 nemesis was a core thing of the spec, the gaming experience and the foundness I have towards the spec/class.
To be more specific, pre-casting Abolish, cating pw:s for the mspeed and then casting Cure along with abolish second's tick basically ensured I'd not be crippled as i ran on 160% mspeed from rogues.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills** 

https://wotlk.evowow.com/?spell=552
https://wowgaming.altervista.org/aowow/?spell=552
https://www.wowhead.com/spell=64129/body-and-soul

Thank you in advance (unless you won't fix it then I revoke my thanks)

**Comment by sherndy** [comment]
The most astute amongst you might find a comment on the same wowhead page disclaiming my point. But take note that it was posted, and later edited on 3.0.9, he is talking about PTR behaviour as the talent was added on "Patch 3.1.0 (14-Apr-2009): Added.", so claims the http://wowwiki.wikia.com/wiki/Body_and_Soul page.

**Comment by MichaelCrow** [comment]
I linked the fix and the doofuses marked it as a duplicate and ignore it lmao.

#Blizzlike

**Comment by Chenocide** [comment]
> I linked the fix and the doofuses marked it as a duplicate and ignore it lmao.
> 
> #Blizzlike

Any normal person would just link the fix in the original report, not sure why that was too hard.

Here's the "fix", the whole priest c++ file from TrinityCore.

https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Spells/spell_priest.cpp

**Comment by MichaelCrow** [comment]
amazing, isnt it? almost like merely updating trinitycore fixes most of your issues or something.

Lets not pretend this isnt what Im talking about:

`-64127 - Body and Soul
class spell_pri_body_and_soul : public SpellScriptLoader
{
    public:
        spell_pri_body_and_soul() : SpellScriptLoader("spell_pri_body_and_soul") { }

        class spell_pri_body_and_soul_AuraScript : public AuraScript
        {
            PrepareAuraScript(spell_pri_body_and_soul_AuraScript);

            bool Validate(SpellInfo const* /*spellInfo*/) override
            {
                return ValidateSpellInfo(
                {
                    SPELL_PRIEST_BODY_AND_SOUL_POISON_TRIGGER,
                    SPELL_PRIEST_ABOLISH_DISEASE
                });
            }

            void HandleProcTriggerSpell(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
            {
                // Proc only on Power Word: Shield
                SpellInfo const* spellInfo = eventInfo.GetSpellInfo();
                if (!spellInfo || !(spellInfo->SpellFamilyFlags[0] & 0x00000001))
                {
                    PreventDefaultAction();
                    return;
                }
            }

            void HandleProcDummy(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
            {
                PreventDefaultAction();

                // Proc only on self casted abolish disease
                SpellInfo const* spellInfo = eventInfo.GetSpellInfo();
                if (!spellInfo)
                    return;

                Unit* caster = eventInfo.GetActor();
                if (spellInfo->Id != SPELL_PRIEST_ABOLISH_DISEASE || caster != eventInfo.GetProcTarget())
                    return;

                if (roll_chance_i(aurEff->GetAmount()))
                    caster->CastSpell(caster, SPELL_PRIEST_BODY_AND_SOUL_POISON_TRIGGER, aurEff);
            }

            void Register() override
            {
                OnEffectProc += AuraEffectProcFn(spell_pri_body_and_soul_AuraScript::HandleProcTriggerSpell, EFFECT_0, SPELL_AURA_PROC_TRIGGER_SPELL);
                OnEffectProc += AuraEffectProcFn(spell_pri_body_and_soul_AuraScript::HandleProcDummy, EFFECT_1, SPELL_AURA_DUMMY);
            }
        };

        AuraScript* GetAuraScript() const override
        {
            return new spell_pri_body_and_soul_AuraScript();
        }
};`

**Comment by sherndy** [comment]
memes

**Comment by sherndy** [comment]
since ppl get responses from bumping their topics imma flood this, been 7 months dude posted the fix
can we just apply it on the very next update pls ty bye. i want to pvp reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

**Comment by sherndy** [comment]
@Lothloryen @Golganeth

**Comment by sherndy** [comment]
@robinsch

**Comment by sherndy** [comment]
this is way more gamebreaking than seduction bypassing cloak on a very especific situation i demand reparations

**Comment by Lothloryen** [comment]
Hello there,

I will look into it after few things I am currently working on are finished. Nonetheless, the suggested code does not fix it, the issue is not in the spell script but in the proc system, thank you for the collaboration though.

**Comment by sherndy** [comment]
still trying to play this priest, been over a year, still mad at spells not working
